File: macroslib/tests/expectations/foreign_interface_cpp_return_not_void.rs
foreign_callback!(callback SomeObserver {
    self_type SomeTrait;
    onStateChanged = SomeTrait::on_state_changed(&self, _: i32, _: bool) -> bool;
    onStateChangedWithoutArgs = SomeObserver::on_state_changed_without_args(&self);
});

File: macroslib/tests/expectations/foreign_interface_cpp_return_not_void.cpp
"virtual bool onStateChanged(int32_t a0, bool a1) const noexcept = 0;";
"virtual void onStateChangedWithoutArgs() const noexcept = 0;";
r#"struct C_SomeObserver {
    void *opaque;
    //! call by Rust side when callback not need anymore
    void (*C_SomeObserver_deref)(void *opaque);

    char (*onStateChanged)(int32_t a0, char a1, void *opaque);

    void (*onStateChangedWithoutArgs)(void *opaque);

};"#;
